Smt/Data/Sexp.lean

```diff
@@ -83,9 +83,9 @@ partial def parseOneAux : List Substring → Except ParseError (Sexp × List Sub
       return (atom tk.toString, tks)
   | [] => throw <| .incomplete "expected a token, got none"
 
-partial def parseManyAux :=
+partial def parseManyAux : List Substring → Except ParseError (Array Sexp × List Substring) :=
```

Comment:
Shifting the type annotation up from `go` to here helps resolves error messages involved with type inference / checking, which we get with `4.9.0`.